chore(docs): Add 21yunbox for deployment (#24429)
gatsbyjs/gatsby: docs/docs/deploying-to-21yunbox.md

---
title: Deploying to 21YunBox
---

> [21YunBox](https://www.21yunbox.com) provides blazing fast Chinese CDN, Continuous Deployment, one click HTTPS and [much more](https://www.21yunbox.com/docs/v2/index.html), providing developers a hassle-free solution to launch their web projects in China.

Author: LGTM

> [21YunBox](https://www.21yunbox.com) provides blazing fast Chinese CDN, Continuous Deployment, one click HTTPS and [much more](https://www.21yunbox.com/docs/v2/index.html), providing developers a hassle-free solution to launch their web projects in China.

Reviewer: Can you add the services, I used "other services like ..... " as a placeholder.

> [21YunBox](https://www.21yunbox.com) provides blazing fast Chinese CDN, Continuous Deployment, one click HTTPS and [other services like ..... ](https://www.21yunbox.com/docs/v2/index.html), providing developers a hassle-free solution to launch their web projects in China.

Reviewer: Hey can you add the other features. as a replacement for "....... "

Author: Oh, didn't see that earlier.

Sure. replace "..." with "managed databases and backend web services".

Should I recommit it?

> Static sites are **insanely easy** on 21YunBox and include the following features:

Reviewer: Again per the [style guide for documentation ](https://www.gatsbyjs.org/contributing/gatsby-style-guide/#avoid-easy-and-simple) we aviod using words like ,"easy" , "simple","insane". More inclusive language is advised 
```suggestion
Static sites are **quick to deploy** on 21YunBox and include the following features:
```

Author: LGTM

> Static sites are **quick to deploy** on 21YunBox and include the following features:

Reviewer 2: ```suggestion
21YunBox includes the following features:
```

- Continuous, automatic builds & deploys from GitHub and Gitee.
- Automatic SSL certificates through [Let's Encrypt](https://letsencrypt.org).
- Instant cache invalidation with a blazing fast, Chinese CDN.
- Unlimited [custom domains](https://www.21yunbox.com/docs/v2/custom-domains.html).
- Automatic [Brotli compression](https://en.wikipedia.org/wiki/Brotli) for faster sites.
- Native HTTP/2 support.
- Automatic HTTP → HTTPS redirects.
- Custom URL redirects and rewrites.

> 

Reviewer 2: Could you remove the periods at the end of these list items, please?

Author: Sure, already removed the periods on the latest commit.

## Prerequisites

> This guide assumes you already have a Gatsby project to deploy. If you need a project, use the [Quick Start](/docs/quick-start) or fork 21YunBox's [Gatsby Example](https://github.com/tobyglei/gatsby) before continuing.

Reviewer: ```suggestion
This guide assumes you already have a Gatsby project to deploy. If you need a project, use the [quick Start](/docs/quick-start) or fork 21YunBox's [Gatsby Example](https://github.com/tobyglei/gatsby) before continuing.
```

Author: LGTM

> This guide assumes you already have a Gatsby project to deploy. If you need a project, use the [quick Start](/docs/quick-start) or fork 21YunBox's [Gatsby Example](https://github.com/tobyglei/gatsby) before continuing.

Reviewer: ```suggestion
This guide assumes you already have a Gatsby project to deploy. If you need a project, use the [quick Start](/docs/quick-start) to get started or fork 21YunBox's [Gatsby Example](https://github.com/tobyglei/gatsby) before continuing.
```

Author: Sure, looks great.

## Setup

You can set up a Gatsby site on 21YunBox in two quick steps:

> 1. Create a new **Web Service** on 21YunBox, and give 21YunBox permission to access your GitHub or Gitee repo.

Reviewer 2: ```suggestion
1. Create a new web service on 21YunBox, and give 21YunBox permission to access your GitHub or Gitee repo.
```

I'd prefer to format text like this only when necessary.

2. Use the following values during creation:

   |                       |                                                  |
   | --------------------- | ------------------------------------------------ |
   | **Environment**       | `Static Site`                                    |
   | **Build Command**     | `yarn && yarn build` (or your own build command) |
   | **Publish Directory** | `./public` (or your own output directory)        |

That's it! Your site will be live on your 21YunBox URL (which looks like `yoursite.21yunbox.com`) as soon as the build is done.

## Continuous deploys

> Now that 21YunBox is connected to your repo, it will **automatically build and publish your site** any time you push to GitHub.

Reviewer 2: ```suggestion
Now that 21YunBox is connected to your repo, it will automatically build and publish your site any time you push to GitHub.
```

## 21YunBox CDN and cache invalidation

> 21YunBox hosts your site on a Chinese, blazing fast CDN which ensures the fastest possible download times for all your users across in China.

Reviewer 2: ```suggestion
21YunBox hosts your site on a Chinese, blazing fast CDN which ensures the fastest possible download times for all your users across China.
```
